chainerx_cc/chainerx/cuda/cuda_device/indexing.cu

```diff
@@ -267,6 +268,33 @@ public:
 
 CHAINERX_CUDA_REGISTER_KERNEL(AddAtKernel, CudaAddAtKernel);
 
+template <typename T>
+struct WhereImpl {
+    using CudaType = cuda_internal::DataType<T>;
+    __device__ void operator()(int64_t /*i*/, bool condition, CudaType x, CudaType y, CudaType& out) { out = condition ? x : y; }
+};
+
+class CudaWhereKernel : public WhereKernel {
+public:
+    void Call(const Array& condition, const Array& x, const Array& y, const Array& out) override {
+        Device& device = x.device();
+        device.CheckDevicesCompatible(condition, x, y, out);
+        const Array& condition_cast = condition.dtype() != Dtype::kBool ? condition.AsType(Dtype::kBool) : condition;
+
+        Dtype out_dtype = ResultType(x, y);
```

Comment:
Could you fix as follows?
```suggestion
        Dtype out_dtype = out.dtype();
```

Reply:
Ahh. Right. On it.

Reply:
If the device functions are called from `chainerx::Where`, this fix does not change the behavior. However, if not, `x` and `y` seems to be better to be casted to `out.dtype`.

Reply:
True, this is more consistent, as in the case if someone calls `out` with a different `dtype` than inferred from `ResultType(x, y)`, that can be a problem.